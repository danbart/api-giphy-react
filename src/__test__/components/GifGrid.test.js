const { shallow } = require("enzyme");
const { GifGrid } = require("../../components/GifGrid");
const React = require("react");
const { useFetchGifs } = require("../../hooks/useFetchGifs");
jest.mock("../../hooks/useFetchGifs");
import "@testing-library/jest-dom";

describe("Pruebas de GifGrid", () => {
  const categoria = "OnePunch";

  test("muestra correctamente", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={categoria} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar items cuando se carga", () => {
    const gifs = [
      {
        id: "abc",
        title: "Cualquier cosa",
        url: "http://localhost/cualquier/cosa.jpg",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={categoria} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
