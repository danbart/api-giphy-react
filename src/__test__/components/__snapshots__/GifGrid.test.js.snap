// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifGrid debe mostrar items cuando se carga 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    OnePunch
  </h3>
  <div
    className="card-grid"
  >
    <ol>
      <GifGridItem
        id="abc"
        key="abc"
        title="Cualquier cosa"
        url="http://localhost/cualquier/cosa.jpg"
      />
    </ol>
  </div>
</Fragment>
`;

exports[`Pruebas de GifGrid muestra correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    OnePunch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;
